hipcheck/src/policy_exprs/token.rs

```diff
@@ -70,6 +73,20 @@ fn lex_ident(input: &mut Lexer<'_, Token>) -> Result<String> {
 	Ok(input.slice().to_owned())
 }
 
+/// Lex a JSON Pointer.
+fn lex_json_pointer(input: &mut Lexer<'_, Token>) -> Result<String> {
+	// Remove the initial '$' character.
+	// Slice safety: the slice is guaranteed to be at least 1 character long,
+	// because the regex for JSONPointer contains a mandatory '$' character.
+	let s = &input.slice()[1..];
```

Comment:
I'm happy to improve panic safety, but I'm not sure how to handle this.

It looks like I could use `get(1..).ok_or(some_error)?` instead, but I'm not sure what error to return. I could fall back to `LexingError::UnknownError`, or add a new variant to `LexingError`. I'd generally be happy to add a new error variant, but I could only imagine calling this `InvalidLexerState`, or `InternalError`, or `Unreachable`. If this is actually what you had in mind, I'll add it.

Mostly I'm confused because it seems impossible to me that `lex_json_pointer` could be called with a slice that doesn't contain an initial `$` character, because of the regex. I could only see that happening due to a programmer error (which I would expect to be caught by the tests).

Reply:
Yeah, this would basically be the moral equivalent of a Rust Internal Compiler Error (ICE). Something fatal that indicates something has gone horribly wrong. In an ideal world we could use something like my crate `woah` to separate fatal errors from recoverable ones, but without a stable `Try` trait we can't do that ergonomically. https://docs.rs/woah/latest/woah/

Reply:
Ok, fair enough! What name would you prefer to use for this type or error?

Reply:
I've pushed a new version using `LexingError::InternalError`.